packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -209,6 +210,24 @@ class UpgradeCommandRunner {
     await runDoctor();
     // Force the welcome message to re-display following the upgrade.
     persistentToolState.setShouldRedisplayWelcomeMessage(true);
+    if (globals.flutterVersion.channel == 'master' || globals.flutterVersion.channel == 'main') {
+      globals.printStatus(
+        '\n'
+        'This channel is intended for Flutter contributors. '
+        'This channel is not as thoroughly tested as the "beta" and "stable" channels. '
+        'We do not recommend using this channel for normal use as it more likely to contain serious regressions.\n'
+        '\n'
+        'If you would like to contribute to Flutter, '
+        'you may need to set up a different development environment. '
```

Comment:
As a casual reader, I didn't understand this statement. I thought `main` _was_ the channel we wanted them to be on? https://github.com/flutter/flutter/wiki/Flutter-build-release-channels says, "We do not recommend using this branch unless [you are contributing to Flutter](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md)."

Reply:
when you contribute to flutter/flutter you usually end up using a fork with our repo as upstream and your fork as origin (at least, that's what we currently recommended in the contributing docs). You still use the master branch as the development branch.

Reply:
The other approach that I've seen recommended is to clone from flutter/flutter and then have a downstream branch. But all-in-all, perhaps this whole sentence is a non-sequitur. Changing channels is orthogonal to setting up a development environment? The other problem about this sentence is that it conflates "contributing to Flutter" with "contributing a pull request to the flutter/flutter repo", which of course aren't the same thing. 

Perhaps we should remove this sentence?

Reply:
Yeah, that's fair. I'll remove it.